src/order/conditionally_complete_lattice.lean

```diff
@@ -338,46 +346,35 @@ le_cSup H (mem_range_self _)
 /--The indexed infimum of two functions are comparable if the functions are pointwise comparable-/
 lemma cinfi_le_cinfi {f g : ι → α} (B : bdd_below (range f)) (H : ∀x, f x ≤ g x) :
   infi f ≤ infi g :=
-begin
-  classical, by_cases hι : nonempty ι,
-  { have Rg : (range g).nonempty, { exactI range_nonempty _ },
-    apply le_cInf Rg,
-    rintros y ⟨x, rfl⟩,
-    have : f x ∈ range f := ⟨x, rfl⟩,
-    exact cInf_le_of_le B this (H x) },
-  { have Rf : range f = ∅, from range_eq_empty.2 hι,
-    have Rg : range g = ∅, from range_eq_empty.2 hι,
-    unfold infi, rw [Rf, Rg] }
-end
+@csupr_le_csupr (order_dual α) _ _ _ _ B H
 
 /--The indexed minimum of a function is bounded below by a uniform lower bound-/
 lemma le_cinfi [nonempty ι] {f : ι → α} {c : α} (H : ∀x, c ≤ f x) : c ≤ infi f :=
-le_cInf (range_nonempty f) (by rwa forall_range_iff)
+@csupr_le (order_dual α) _ _ _ _ _ H
 
 /--The indexed infimum of a function is bounded above by the value taken at one point-/
 lemma cinfi_le {f : ι → α} (H : bdd_below (range f)) (c : ι) : infi f ≤ f c :=
-cInf_le H (mem_range_self _)
-
-@[simp] theorem cinfi_const [hι : nonempty ι] {a : α} : (⨅ b:ι, a) = a :=
-by rw [infi, range_const, cInf_singleton]
+@le_csupr (order_dual α) _ _ _ H c
 
 @[simp] theorem csupr_const [hι : nonempty ι] {a : α} : (⨆ b:ι, a) = a :=
 by rw [supr, range_const, cSup_singleton]
 
-theorem infi_unique [unique ι] {s : ι → α} : (⨅ i, s i) = s (default ι) :=
-have ∀ i, s i = s (default ι) := λ i, congr_arg s (unique.eq_default i),
-by simp only [this, cinfi_const]
+@[simp] theorem cinfi_const [hι : nonempty ι] {a : α} : (⨅ b:ι, a) = a :=
+@csupr_const (order_dual α) _ _ _ _
 
 theorem supr_unique [unique ι] {s : ι → α} : (⨆ i, s i) = s (default ι) :=
 have ∀ i, s i = s (default ι) := λ i, congr_arg s (unique.eq_default i),
 by simp only [this, csupr_const]
 
-@[simp] theorem infi_unit {f : unit → α} : (⨅ x, f x) = f () :=
-by { convert infi_unique, apply_instance }
+theorem infi_unique [unique ι] {s : ι → α} : (⨅ i, s i) = s (default ι) :=
+@supr_unique (order_dual α) _ _ _ _
 
 @[simp] theorem supr_unit {f : unit → α} : (⨆ x, f x) = f () :=
 by { convert supr_unique, apply_instance }
 
+@[simp] theorem infi_unit {f : unit → α} : (⨅ x, f x) = f () :=
+@supr_unit (order_dual α) _ _
+
```

Comment:
I reordered these slightly for consistency with the lemmas above, putting the `sup` versions before the `inf` versions.